src/Microsoft.DotNet.Build.CloudTestTasks/FilterForUpload.cs

```diff
@@ -0,0 +1,93 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+namespace Microsoft.DotNet.Build.CloudTestTasks
+{
+    public sealed class FilterForUpload : Task
+    {
+        [Required]
+        public string TestOutputDirListsLocation { get; set; }
+
+
+        [Required]
+        public string FilterToTestTFM { get; set; }
+
+        [Required]
+        public string FilterToOSGroup { get; set; }
+
+        [Output]
+        public ITaskItem[] FilteredUploadItems
+        {
+            get { return _copyLocalItems.ToArray(); }
+        }
+
+        private readonly List<ITaskItem> _copyLocalItems = new List<ITaskItem>();
+
+        public override bool Execute()
+        {
+
+            Debugger.Launch();
+            string[] filesToSearch = Directory.GetFiles(TestOutputDirListsLocation);
+            
+            foreach (var filename in filesToSearch)
+            {
+                string selectedTest = string.Empty;
+                using (StreamReader sr = new StreamReader(File.OpenRead(filename)))
+                {
+                    string line;
+                    while ((line = sr.ReadLine()) != null)
+                    {
+                        if (calculateMatchValue(line) > calculateMatchValue(selectedTest))
+                        {
+                            int lastIndexOfSlash = line.LastIndexOf('/');
+                            string path = line.Substring(0, lastIndexOfSlash);
+                            string testTfm = line.Substring(lastIndexOfSlash + 1);
```

Comment:
It's also a little troublesome that we are requring a specific path format (namely that testTfm is the second to last path fragment in the line in this file who's format isn't documented).  It seems fragile, and will make updating this later very difficult.